OPHD/States/MapViewStateTurn.cpp

```diff
@@ -350,6 +351,34 @@ void MapViewState::updateFood()
 }
 
 
+/**
+ * Update road intersection patterns
+ */
+void MapViewState::updateRoads()
+{
+	auto roads = NAS2D::Utility<StructureManager>::get().structureList(Structure::StructureClass::Road);
+
+	for (auto road : roads)
+	{
+		if (!road->operational()) { continue; }
+
+		const auto tileLocation = NAS2D::Utility<StructureManager>::get().tileFromStructure(road).position();
+
+		std::array<bool, 4> surroundingTiles{ false, false, false, false };
+		for (size_t i = 0; i < 4; ++i)
+		{
+			const auto tileToInspect = tileLocation + DirectionClockwise4[i];
+			if (!mTileMap->isValidPosition(tileToInspect)) { continue; }
+			if (!mTileMap->getTile(tileToInspect).thingIsStructure()) { continue; }
+
+			surroundingTiles[i] = mTileMap->getTile(tileToInspect).structure()->structureId() == StructureID::SID_ROAD;
+		}
```

Comment:
One more comment I forgot to add:

In other projects, I've seen this sort of thing tackled where the bit flags are binary encoded into an index. Effectively there are 16 possible graphics for 4 possibly adjacent tiles, including T-intersections.

```cpp
int index = 0;
int field = 1;
for (const auto direction : DirectionClockswise4)
{
  if (check(tileLocation + direction))
  {
    index |= field;
    field <<= 1;
  }
}
```

If you exclude T-intersections, then there are 4 possible graphics.

```cpp
const int index =
  (check(tilePosition + DirectionNorth) | check(tilePosition + DirectionSouth)) |
  (check(tilePosition + DirectionEast) | check(tilePosition + DirectionWest)) * 2;
```

An index of 0 means no adjacent tiles. Assuming no unconnected point graphics, you can probably default index 0 to some default direction. A max index (either 15 or 3), means all adjacent tiles (cross intersection).

The index can then potentially be used directly to select the tile index to draw from an array of images.

Reply:
That would probably be a faster method that uses less memory but it's also a little less understandable when quickly reading through it. I opted for pattern matching in this case because it's pretty simple and there are a very finite number of options.

Something to consider for the future though.